### PR TITLE
GLSL preprocessor output preview.

### PR DESCRIPTION
**Pull request by dor666**
I needed to debug some shaders using preprocessor. It was quite easy to add this functionality to this plugin. Let me know what you think about it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Or Settings/Preferences ➔ Packages ➔ Search for `preview`
 - [x] [EmberScript](https://github.com/Glavin001/atom-preview/issues/39)
 - [x] [ng-classify](https://github.com/Glavin001/atom-preview/issues/60)
 - [x] [LiveScript](https://github.com/Glavin001/atom-preview/issues/89)
+- [x] glslPreprocessor
```
Comment (nemoDreamer) [inline README.md]: I created #145. Could you link to it in the `README`?

File: lib/renderer.coffee [hunk 2]
```diff
@@ -284,3 +284,27 @@ module.exports =
...
       lang: -> 'json'
+    'glsl':
+      render: (text, filepath, cb) ->
+        try
+          # glslangValidator needs file not text, so we create temp file.
+          # extension must be the same, so validator can detect shader type.
+          tmp = require 'tmp'
+          extension = '.' + filepath.split('.').pop();
+          tempFile = tmp.fileSync {prefix: 'atom-preview-editor', postfix: extension};
+          file = fs.writeFileSync tempFile.fd, text
```
Comment (nemoDreamer) [inline lib/renderer.coffee]: Would be sweet if you could use `file`/`writeFile` instead, so as to not block the thread. Thanks to `cb`, all should still work as expected.

**Review by nemoDreamer — CHANGES_REQUESTED** [review]
Hey @dor666 , great addition, thanks!
Would you mind making two small changes?

**Comment by nemoDreamer** [comment]
Oh: one more thing: mind adding a quick little example shader into `spec/samples`? Thanks!

**Comment by dor666** [comment]
Hi @nemoDreamer.
I've added the changes you mentioned and fixed additional spotted bugs.
Could you take a look?

File: README.md [hunk 3]
```diff
@@ -53,6 +54,10 @@ For instance, if you type or switch tabs,
 how long of a pause before the preview refreshes.
 - `Update On Tab Change` (boolean) -
 Should the preview update to the currently active tab?
+- `glslangValidator` (string) - Path to [glslangValidator](https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/) . Needed for [GLSL](https://en.wikipedia.org/wiki/OpenGL_Shading_Language) preview (pre-processed). You can compile it from [source](https://github.com/KhronosGroup/glslang) or download as part of [VulkanSDK](https://www.lunarg.com/vulkan-sdk/).
```
Comment (nemoDreamer) [inline README.md]: Isn't this gonna read `Glslang Validator`?

File: README.md [hunk 4]
```diff
@@ -53,6 +54,10 @@ For instance, if you type or switch tabs,
...
+- `Remove Extra Blank Lines` (boolean) - Remove extra blank lines in glsl preprocessed output.
+
+removeExtraBlankLines
```
Comment (nemoDreamer) [inline README.md]: Left-over?

**Review by nemoDreamer — APPROVED** [review]
Noice!
(One little comment about the wording in the settings)

File: README.md (hunk 3 above)
Comment (nemoDreamer) [reply README.md]: Also typo: remove the space between the `)` and the `.`

File: README.md [hunk 5]
```diff
@@ -53,6 +54,10 @@ For instance, if you type or switch tabs,
 how long of a pause before the preview refreshes.
 - `Update On Tab Change` (boolean) -
 Should the preview update to the currently active tab?
+- `glslangValidator` (string) - Path to [glslangValidator](https://www.khronos.org/opengles/sdk/tools/Reference-Compiler/) . Needed for [GLSL](https://en.wikipedia.org/wiki/OpenGL_Shading_Language) preview (pre-processed). You can compile it from [source](https://github.com/KhronosGroup/glslang) or download as part of [VulkanSDK](https://www.lunarg.com/vulkan-sdk/).
+- `Remove Extra Blank Lines` (boolean) - Remove extra blank lines in glsl preprocessed output.
```
Comment (nemoDreamer) [inline README.md]: And I guess while you're in there, capitalize `glsl`?

File: lib/preview.coffee [hunk 6]
```diff
@@ -12,6 +12,12 @@ module.exports =
     liveUpdate:
       type: 'boolean'
       default: true
+    glslangValidator:
```
Comment (nemoDreamer) [inline lib/preview.coffee]: Would a different human-readable setting name convert better to title case?
`GLSLValidator -> GLSL Validator`

File: README.md (hunk 4 above)
Comment (dor666) [reply README.md]: Yes, thanks.

File: README.md (hunk 5 above)
Comment (dor666) [reply README.md]: Yes, correct.

**Comment by dor666** [comment]
Hi, is there still something to polish before it will be merged?

**Comment by nemoDreamer** [comment]
Thanks for the addition, @dor666 !

**Pull request merged**